core/server/server.js

```diff
@@ -494,65 +488,65 @@ class Server {
 
     log.log(`Server is starting up: ${this.baseUrl}`)
 
-    const camp = (this.camp = Camp.create({
-      documentRoot: this.config.public.documentRoot,
-      port,
-      hostname,
-      secure,
-      staticMaxAge: 300,
-      cert,
-      key,
-    }))
+    const app = (this.app = express())
 
     if (requireCloudflare) {
       this.requireCloudflare()
     }
 
     const { githubConstellation, metricInstance } = this
-    await githubConstellation.initialize(camp)
+    await githubConstellation.initialize(app)
     if (metricInstance) {
       if (this.config.public.metrics.prometheus.endpointEnabled) {
-        metricInstance.registerMetricsEndpoint(camp)
+        metricInstance.registerMetricsEndpoint(app)
       }
       if (this.influxMetrics) {
         this.influxMetrics.startPushingMetrics()
       }
     }
 
     const { apiProvider: githubApiProvider } = this.githubConstellation
-    setRoutes(allowedOrigin, githubApiProvider, camp)
+    setSuggestRoutes(allowedOrigin, githubApiProvider, app)
 
     // https://github.com/badges/shields/issues/3273
-    camp.handle((req, res, next) => {
+    app.use((req, res, next) => {
       res.setHeader('Access-Control-Allow-Origin', '*')
       next()
     })
 
     this.registerErrorHandlers()
     this.registerRedirects()
+    app.use(
+      express.static(this.config.public.documentRoot, {
+        // Since express's `maxAge` parameter sets `Cache-Control: public`, set
+        // the headers manually insetad.
+        cacheControl: false,
+        setHeaders: res =>
+          res.setHeader('Cache-Control', 'max-age=300, s-maxage=300'),
+      })
+    )
     await this.registerServices()
+    if (registerExtras) {
+      registerExtras(app)
+    }
+    this.registerNotFoundHandlers()
 
-    camp.timeout = this.config.public.requestTimeoutSeconds * 1000
-    if (this.config.public.requestTimeoutSeconds > 0) {
-      camp.on('timeout', socket => {
-        const maxAge = this.config.public.requestTimeoutMaxAgeSeconds
-        socket.write('HTTP/1.1 408 Request Timeout\r\n')
-        socket.write('Content-Type: text/html; charset=UTF-8\r\n')
-        socket.write('Content-Encoding: UTF-8\r\n')
-        socket.write(`Cache-Control: max-age=${maxAge}, s-maxage=${maxAge}\r\n`)
-        socket.write('Connection: close\r\n\r\n')
-        socket.write('Request Timeout')
-        socket.end()
-      })
+    if (secure) {
+      this.server = https.createServer({ hostname, cert, key }, app)
+    } else {
+      this.server = http.createServer({ hostname }, app)
     }
-    camp.listenAsConfigured()
 
-    await new Promise(resolve => camp.on('listening', () => resolve()))
+    this.server.setTimeout(this.config.public.requestTimeoutSeconds * 1000)
```

Comment:
1. If we're going to move to just dropping the connection on timeout, I think we can just get rid of the `requestTimeoutMaxAgeSeconds` setting as it won't do anything
2. `requestTimeoutSeconds` is an optional setting (for self hosting users). We should handle the case where it is not set.